sim-lib/src/lib.rs

```diff
@@ -435,6 +447,23 @@ enum SimulationOutput {
     SendPaymentFailure(Payment, PaymentResult),
 }
 
+type MutRngType = Arc<StdMutex<Box<dyn RngCore + Send>>>;
```

Comment:
`StdMutex` so that we don't need to worry about making this trait `async`? Let's add a comment here, perhaps we want to go down that refactor road one day, and it would be good to remember why we chose this.